rllib/policy/sample_batch.py

```diff
@@ -1619,8 +1623,20 @@ def concat_samples_into_ma_batch(samples: List[SampleBatchType]) -> "MultiAgentB
     return MultiAgentBatch(out, env_steps)
 
 
-def _concat_key(*values, time_major=None):
-    return concat_aligned(list(values), time_major)
+def _concat_values(*values, time_major=None) -> TensorType:
+    """Concatenates a list of values.
+
+    Args:
+        values: The values to concatenate.
+        time_major: Whether to concatenate along the first axis
+            (time_major=False) or the second axis (time_major=True).
+    """
+    if torch and any([torch.is_tensor(t) for t in values]):
+        # Incase the elements are torch values, they might be on GPU, so we have to
+        # use torch.cat here.
+        return torch.cat(list(values), dim=1 if time_major else 0)
+    else:
+        return np.concatenate(list(values), axis=1 if time_major else 0)
```

Comment:
This is the general purpose numpy concatenate that we used before.
This is on the critical path of all ordinary use cases.

torch.cat is only for the special case of users using torch *GPU* tensors.